include/astarte_bson.h

```diff
@@ -20,19 +24,162 @@
 extern "C" {
 #endif
 
-const void *astarte_bson_key_lookup(const char *key, const void *document, uint8_t *type);
-void *astarte_bson_next_item(const void *document, const void *current_item);
-const void *astarte_bson_first_item(const void *document);
-const char *astarte_bson_key(const void *item);
-const char *astarte_bson_value_to_string(const void *value_ptr, uint32_t *len);
-const char *astarte_bson_value_to_binary(const void *value_ptr, uint32_t *len);
-const void *astarte_bson_value_to_document(const void *value_ptr, uint32_t *len);
-int8_t astarte_bson_value_to_int8(const void *value_ptr);
-int32_t astarte_bson_value_to_int32(const void *value_ptr);
-int64_t astarte_bson_value_to_int64(const void *value_ptr);
-double astarte_bson_value_to_double(const void *value_ptr);
-bool astarte_bson_check_validity(const void *document, unsigned int file_size);
-uint32_t astarte_bson_document_size(const void *document);
+/**
+ * @brief Look up the key from the document and return a pointer to the appropriate entry.
+ *
+ * @details This function loops over all the elements in the document's list and return a pointer
+ * to the first element with a name matching the specified key.
+ * N.B. The return value points to the content of the element, past the element type and name.
+ *
+ * @note Deprecated, call astarte_bson_deserializer_element_lookup() instead.
+ *
+ * @param[in] key String representing the key to find in the document.
+ * @param[in] document Document to use for the search.
+ * @param[out] type Returned element type, see 'astarte_bson_types.h' for its possible values.
+ * @return Pointer to the content of the element matched by the key. NULL when the key has not been
+ * matched or the BSON file is malformed.
+ */
+const void *astarte_bson_key_lookup(const char *key, const void *document, uint8_t *type)
+    __attribute__((
+        deprecated("Please use the deserialization functions in astarte_bson_deserializer.h")));
```

Comment:
What do you think about suggesting the new function in the depracated log as well?
```suggestion
        deprecated("Please use the 'astarte_bson_deserializer_element_lookup' function in astarte_bson_deserializer.h")));
```

Reply:
Sure